src/components/vendorcomponents/InputWithLabel.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+interface TextInputProps {
+    label: string;
+    register: any;
+    name: string;
+}
+
+export const InputWithLabel: React.FC<TextInputProps> = ({
+    label,
+    register,
+    name,
+}) => (
+    <div>
+        <label className="block text-sm font-medium text-gray-700">{label}</label>
+        <input
+            {...register(name)}
+            className="mt-1 block w-full border border-gray-300 rounded-md shadow-sm p-1"
+        />
+    </div>
```

Comment:
You're still repeating yourself here. make one component that reusable, not making multiple similar components